sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,220 @@
+
+[source,c++]
+----
+// Load API
+template <typename T, access::address_space Space, access::decorated IsDecorated>
+T load(const multi_ptr<T, Space, IsDecorated>* src);
+
+template <int N, typename T, access::address_space Space, access::decorated IsDecorated>
+vec<T, N> load(const multi_ptr<T, Space, IsDecorated> src);
+
+// Store API
+template <typename T, access::address_space Space, access::decorated IsDecorated>
+void store(multi_ptr<T, Space, IsDecorated> dst, const T& x);
+
+template <typename T, access::address_space Space, access::decorated IsDecorated>
+void store(multi_ptr<T, Space, IsDecorated> dst, const vec<T, N>& x);
+----
+
+== Extended Proposal
+
+The extended APIs provides the following features:
+
+* load/store with temporary memory buffer, which can be passed via GroupHelper such as `group_with_scratchpad` link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc#group-helper[group_with_scratchpad description from sort over group proposal]
+
+* load/store flexible amount of elements per work item as sycl::span
+
+* specify data placement type via properties (1) or group_memory_helper (*naming is TBD) class (5)
+
+* extra options such as setting boundry values or limiting numbers of work items can be also specified via group_memory_helper
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Load API
+template<typename GroupHelper, typename InputIterator, typename OutputT,
+        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
+void load_over_group(GroupHelper gh, InputIteratorT in_ptr,
+                      sycl::span<OutputT, ItemsPerWorkItem> out, Properties = {}); (1)
+
+template<typename GroupHelper, typename InputIterator, typename OutputT,
+        std::size_t ElementsPerWorkItem, typename GroupMemoryHelper>
+void load_over_group(GroupHelper gh, InputIteratorT in_ptr,
+                      sycl::span<OutputT, ItemsPerWorkItem> out,
+                      GroupMemoryHelper<typename InputIterator::value_type, ItemsPerWorkItem> memory_helper = {}); (2)
+
+// Store API
+template<typename GroupHelper, typename OutputIterator, typename InputT,
+        std::size_t ElementsPerWorkItem, typename Properties = properties<>>
+void store_over_group(GroupHelper gh, OutputIterator out_ptr,
+                      sycl::span<InputT, ItemsPerWorkItem> in, Properties = {}); (3)
+
+template<typename GroupHelper, typename OutputIterator, typename InputT,
+                  std::size_t ElementsPerWorkItem, typename GroupMemoryHelper>
+void store_over_group(GroupHelper gh, OutputIterator out_ptr,
+                      sycl::span<InputT, ItemsPerWorkItem> in,
+                      GroupMemoryHelper<InputT, ItemsPerWorkItem> memory_helper = {}); (4)
+
+// Properties:
+enum class group_algorithm_data_placement {
```

Comment:
Bikeshedding: This is a very long name, so we should try and find something shorter.

Reply:
I also haven't yet figured out if ve need input and output data placement properties or just one-way one (as in Example 2)